trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/main_container/MainContainerScreen.kt

```diff
@@ -39,21 +37,31 @@ internal fun MainContainerScreen(
         }
     }
 
-    MainContainerScreenContent(state = state)
+    MainContainerScreenContent(state = state, listener = viewModel)
 }
 
 @Composable
-private fun MainContainerScreenContent(state: MainContainerState) {
-    TrendsAnimatedVisibility(
-        visible = state.isCategoriesAlreadySelectedByUser == null
-    ) {
-        Box(
-            modifier = Modifier
-                .fillMaxSize()
-                .background(Theme.colorScheme.background.surface.copy(alpha = 0.7f)),
-            contentAlignment = Alignment.Center
+private fun MainContainerScreenContent(
+    state: MainContainerState,
+    listener: MainContainerInteractionListener
+) {
+    Scaffold {
+        TrendsAnimatedVisibility(
+            visible = state.isCategoriesAlreadySelectedByUser == null
         ) {
-            DotsProgressIndicator()
+            LoadingProgressBar()
+        }
+
+        TrendsAnimatedVisibility(state.error is ErrorState.RequestFailed) {
+            SomethingWentWrong(
+                onRetry = { listener.onClickRetry() }
```

Comment:
Isuggest use it as we use in other files.

```suggestion
                onRetry =  listener::onClickRetry
```